tests/marqo_test.py

```diff
@@ -266,7 +289,10 @@ def cleanup_documents_from_index(self, index_to_cleanup: str):
             self.client.config.instance_mapping.get_index_base_url(idx.index_name)
 
             if self.index_to_documents_cleanup_mapping.get(index_to_cleanup):
-                idx.delete_documents(list(self.index_to_documents_cleanup_mapping[index_to_cleanup]), auto_refresh=True)
+                res = idx.delete_documents(list(self.index_to_documents_cleanup_mapping[index_to_cleanup]),
+                                           auto_refresh=True)
+                if res['status'] == 'succeeded':
+                    self.index_to_documents_cleanup_mapping[index_to_cleanup] = set()
```

Comment:
Are you sure it is safe to do this? Isn't it safer to iterate through each result in `res`, removing successfully deleted docs from `documents_to_cleanup`?

Reply:
out of 200 executions it didn't fail at all, anyways we have default deletion by search after it so it should be safe!